### PR TITLE
Multiple file parse and download support

### PR DESCRIPTION
**Pull request by stripes416**

### CODE REVIEW

**Comment by tsviet** [comment]
objects response and reader were deleted?
they still in use in other functions be shure to check other functions then you delete variables.

**Comment by tsviet** [comment]
This will make TDD valuable at this point

**Comment by SoySauceLabs** [comment]
working on a fix at this very moment

**Comment by kevbui** [comment]
Using reader and response like that is somewhat hacky. Which is why I removed the ability to use them like that. 

The new way of using currentDirFiles makes the code way simpler.

**Comment by SoySauceLabs** [comment]
K, fix has been applied.

**Comment by tsviet** [comment]
You are wrong. You start to use dublicated code in get function that is used in Constractor. We need to be avle to reuse code.

**Comment by tsviet** [comment]
Now it is in fetchCurrentDirectoryItems() and Get() same code

**Comment by SoySauceLabs** [comment]
Mmm...I'm not sure I see it at the moment. Do you mind applying comments to the lines where you see duplication?

**Comment by kevbui** [comment]
And I somewhat agree that get() needs to be refactored a bit. But that can be down the road.

**Comment by tsviet** [comment]
What is bad about creating object in constractor and reuse then in functions?

**Comment by kevbui** [comment]
I think there is confusion on the use of (Ftp)WebRequest objects . It's a bad practice to keep reusing a request pipeline.

That's why C# will throw werid errors if you try reusing it.

**Comment by SoySauceLabs** [comment]
Mmm...yes I think I see the confusion now too. Each "request" is unique. Similar principle (I believe) to HTTP requests, only we're using an FTP protocol.

**Comment by tsviet** [comment]
Any suggestion for a best practice? How about creating function that return response object and then close it after use?

File: FTPizza/Ftp.cs [hunk 1]
```diff
@@ -157,9 +163,9 @@ private string ParseItem(string item)
                     throw new Exception("Malformatted file: " + item);
                 }
 
-                while ((line = reader.ReadLine()) != null)
-                {
-                    if (line.Contains(item))
+                foreach (string file in currentDirFiles)
```
Comment (kevbui) [inline FTPizza/Ftp.cs]: Instead of foreaching the loop, you can just call currentDirFiles.contains(item)

**Comment by SoySauceLabs** [comment]
Good catch! I think that was a result of how the files were stored in the "reader". It's been removed.

**Pull request merged**